### PR TITLE
UI Component added

### PR DESCRIPTION
**Pull request by ThanyaniDev**
I added two nib files, landing view(entry View), and Weather details view(details View).
Please review and let me know what you think.

### CODE REVIEW

File: MarsMission/Controllers/DetailsViewController.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+//  DetailsViewController.swift
+//  MarsMission
+//
+//  Created by Thanyani on 2020/11/03.
+//
+
+import UIKit
+
+class DetailsViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource{
+
```
Comment (Sashen943) [inline MarsMission/Controllers/DetailsViewController.swift]: Please fix the spacing between lines of code

File: MarsMission/Controllers/DetailsViewController.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
+//
+//  DetailsViewController.swift
+//  MarsMission
+//
+//  Created by Thanyani on 2020/11/03.
+//
+
+import UIKit
+
+class DetailsViewController: UIViewController, UICollectionViewDelegate, UICollectionViewDataSource{
```
Comment (Sashen943) [inline MarsMission/Controllers/DetailsViewController.swift]: Spacing between bracket and UICollectionViewDataSource

File: MarsMission/Controllers/LandingViewController.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+import UIKit
+
+class LandingViewController: UIViewController {
+
+    @IBOutlet weak var dailyWeatherReport: UIButton!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        navigationItem.title = "Mars Mission"
```
Comment (Sashen943) [inline MarsMission/Controllers/LandingViewController.swift]: Localise this string

File: MarsMission/Controllers/ViewController.swift [hunk 4]
```diff
@@ -9,9 +9,11 @@ import UIKit
 
 class ViewController: UIViewController {
 
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        navigationItem.title = "Weather Forecast"
```
Comment (Sashen943) [inline MarsMission/Controllers/ViewController.swift]: Localise this string

File: MarsMission/getData.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+public class GetData {
+    
+    @Published var atmosphericTemperatureData = [AtmosphericTemperatureData]()
+    
+    init() {
+        fecthData()
+    }
+    
+    func fecthData() {
```
Comment (Sashen943) [inline MarsMission/getData.swift]: This fetch data function must contain a completion handler

**Review by Sashen943 — COMMENTED** [review]
Please provide screenshots on this pull request of the views that you have built

File: MarsMission/getData.swift (hunk 5 above)
Comment (ThanyaniDev) [reply MarsMission/getData.swift]: Screenshot 2020-11-05 at 05 52 14

File: MarsMission/Controllers/LandingViewController.swift (hunk 3 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/LandingViewController.swift]: Screenshot 2020-11-05 at 05 55 52

File: MarsMission/Controllers/ViewController.swift (hunk 4 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/ViewController.swift]: Screenshot 2020-11-05 at 05 55 52

File: MarsMission/Controllers/DetailsViewController.swift (hunk 1 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/DetailsViewController.swift]: Screenshot 2020-11-05 at 06 23 51

File: MarsMission/Controllers/DetailsViewController.swift (hunk 2 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/DetailsViewController.swift]: Screenshot 2020-11-05 at 06 23 51

**Comment by ThanyaniDev** [comment]
Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-05 at 06 15 53

**Comment by ThanyaniDev** [comment]
Simulator Screen Shot - iPhone 12 Pro Max - 2020-11-06 at 07 09 13

**Comment by ThanyaniDev** [comment]
I have added the Landing Screen and Dailly weather report screens.

File: MarsMission/Controllers/DetailsViewController.swift [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+//
+//  Created by Thanyani on 2020/11/03.
+//
+
+import UIKit
+
+class DetailsViewController: UICollectionViewController {
+	
+	override func viewDidLoad() {
+		super.viewDidLoad()
```
Comment (Sashen943) [inline MarsMission/Controllers/DetailsViewController.swift:13]: Please fix indentation and spacing

File: MarsMission/Controllers/DetailsViewController.swift [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	
+	
+	func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
+		return .init(width: view.frame.width - 10, height: 200)
+	}
+	
+}
+
+extension DetailsViewController: UICollectionViewDelegateFlowLayout {
+
```
Comment (Sashen943) [inline MarsMission/Controllers/DetailsViewController.swift]: Empty extension?? Are you committing a project that is compiling?

File: MarsMission/Controllers/LandingViewController.swift [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+		super.viewDidLoad()
+		
+		navigationItem.title = keys.navigationTitle.landingtitle
+		dailyWeatherReport.layer.cornerRadius = 30
+		threeDayWeatherReport.layer.cornerRadius = 30
+		seasonalWeatherReport.layer.cornerRadius = 30
+		
+		ServiceImplementation.shared.getWeatherData() { result in
+			switch result {
+				case .success(let results):
```
Comment (Sashen943) [inline MarsMission/Controllers/LandingViewController.swift]: Please fix spacing and indentation

File: MarsMission/Controllers/LandingViewController.swift [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+					print(results)
```
Comment (Sashen943) [inline MarsMission/Controllers/LandingViewController.swift]: Why are we making a service call in the view controller?

**Review by Sashen943 — CHANGES_REQUESTED** [review]

File: MarsMission/Controllers/DetailsViewController.swift (hunk 7 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/DetailsViewController.swift]: Screenshot 2020-11-06 at 10 44 58

File: MarsMission/Controllers/DetailsViewController.swift (hunk 6 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/DetailsViewController.swift:13]: Screenshot 2020-11-06 at 10 44 58

File: MarsMission/Controllers/LandingViewController.swift (hunk 8 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/LandingViewController.swift]: Screenshot 2020-11-06 at 10 50 24

File: MarsMission/Controllers/LandingViewController.swift (hunk 9 above)
Comment (ThanyaniDev) [reply MarsMission/Controllers/LandingViewController.swift]: Screenshot 2020-11-06 at 10 50 24

File: MarsMission/Service/ServiceImplementation.swift [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+				completion(.failure(.dataError))
+				return
+			}
+			
+			guard let response = response as? HTTPURLResponse,  response.statusCode == 200 else {
+				completion(.failure(.responseError))
+				return
+			}
+			
+			if let datas = data {
```
Comment (Sashen943) [inline MarsMission/Service/ServiceImplementation.swift]: if let data = data

File: MarsMission/ViewModels/WeatherDataViewModel.swift [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+					case .success(let weatherdata):
+						self.weatherfecthWeatherDataSuccess(weatherdata)
+					case .failure(let error):
+						self.weatherfecthWeatherDataFail(error)
+				}
+			}
+		}
+	}
+	
+	private func weatherfecthWeatherDataSuccess(_ weatherdata: WeatherData) {
```
Comment (Sashen943) [inline MarsMission/ViewModels/WeatherDataViewModel.swift]: spelling of fetch

Comment (Sashen943) [inline MarsMission/ViewModels/WeatherDataViewModel.swift]: please name functions correctly i.e. handleThatFetchWeatherDataSucceeds

File: MarsMission/ViewModels/WeatherDataViewModel.swift [hunk 12]
```diff
@@ -0,0 +1,47 @@
...
+		DispatchQueue.main.async {
+			self.weatherdata = weatherdata
+			self.view.populateWeatherData("\(weatherdata.mn) NaN° F | C")
+		}
+	}
+	
+	private func weatherfecthWeatherDataFail(_ error: Error) {
+		DispatchQueue.main.async {
```
Comment (Sashen943) [inline MarsMission/ViewModels/WeatherDataViewModel.swift:43]: please name functions correctly i.e. handleThatFetchWeatherDataFails

File: MarsMission/Views/SearchViewCell.swift [hunk 13]
```diff
@@ -0,0 +1,73 @@
...
+//  MarsMission
+//
+//  Created by Thanyani on 2020/11/05.
+//
+
+import UIKit
+
+class SearchViewCell: UICollectionViewCell {
+	
+	private lazy  var viewModel =  WeatherDataViewModel(view: self)
```
Comment (Sashen943) [inline MarsMission/Views/SearchViewCell.swift]: Why is the view model in the cell and not the view controller

**Review by Sashen943 — CHANGES_REQUESTED** [review]

File: MarsMission/Service/ServiceImplementation.swift (hunk 10 above)
Comment (ThanyaniDev) [reply MarsMission/Service/ServiceImplementation.swift]: Screenshot 2020-11-06 at 15 57 51

File: MarsMission/ViewModels/WeatherDataViewModel.swift (hunk 11 above)
Comment (ThanyaniDev) [reply MarsMission/ViewModels/WeatherDataViewModel.swift]: Screenshot 2020-11-06 at 15 59 48

Comment (ThanyaniDev) [reply MarsMission/ViewModels/WeatherDataViewModel.swift]: Screenshot 2020-11-06 at 16 07 32

Comment (ThanyaniDev) [reply MarsMission/ViewModels/WeatherDataViewModel.swift]: Screenshot 2020-11-06 at 16 07 32

File: MarsMission/ViewModels/WeatherDataViewModel.swift (hunk 12 above)
Comment (ThanyaniDev) [reply MarsMission/ViewModels/WeatherDataViewModel.swift:43]: Screenshot 2020-11-06 at 16 10 20

File: MarsMission/Views/SearchViewCell.swift (hunk 13 above)
Comment (ThanyaniDev) [reply MarsMission/Views/SearchViewCell.swift]: Screenshot 2020-11-06 at 16 21 25

File: MarsMission/Models/WeatherData.swift [hunk 14]
```diff
@@ -9,40 +9,17 @@ import Foundation
...
-			if let atcon =  try? continer.nestedContainer(keyedBy: atmosphericTemp.self, forKey: .AT) {
-				
-				self.av = try atcon.decode(Double.self, forKey: .av)
-				self.mn = try atcon.decode(Double.self, forKey: .mn)
-				self.mv = try atcon.decode(Double.self, forKey: .mv)
-			}
-			
-		}
-	}
+	var forecasts: [forecastsData]
```
Comment (Sashen943) [inline MarsMission/Models/WeatherData.swift:11]: Class naming here is incorrect. All classes should start with Capitals. This class should be named Forecast.
`var forecasts: [Forecast]`

**Review by Sashen943 — APPROVED** [review]
Looking good Thanyani. Some small changes need to be made but happy to approve for now

**Pull request merged**